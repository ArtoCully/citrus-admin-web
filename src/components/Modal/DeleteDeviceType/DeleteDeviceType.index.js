// @flow
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { removeDeviceType } from 'src/redux/modules/device-type'
import DeleteDeviceType from './DeleteDeviceType'

type StateProps = {}

function mapStateToProps (
  state: GlobalReducerState,
  ownProps: Object,
): StateProps {

  return {}

}

type DispatchProps = {
  deleteDeviceType: Function,
}

function mapDispatchToProps (dispatch: any): DispatchProps {

  return {
    deleteDeviceType: bindActionCreators(removeDeviceType, dispatch),
  }

}

export default connect(mapStateToProps, mapDispatchToProps)(DeleteDeviceType)
export type ReduxProps = StateProps
