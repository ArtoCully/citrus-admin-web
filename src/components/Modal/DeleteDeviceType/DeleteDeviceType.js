// @flow
import React, { PureComponent } from 'react'
import LinkReplace from 'src/components/shared/LinkReplace/LinkReplace'
import css from './DeleteDeviceType.style.css'
import type { ReduxProps } from './DeleteDeviceType.index'


type Props = ReduxProps & {
  location: Object,
}

type State = {
  location: {
    deviceType: Object
  },
}

class DeleteDeviceType extends PureComponent {

  state: State

  delete = (id: Number) => {

    this.props.deleteDeviceType(id)

  }

  props: Props

  render (): ?React$Element<any> {

    const {
      location: {
        state: {
          deviceType = {},
        },
        pathname,
      },
    } = this.props

    return (
      <div className={css.deleteDeviceType}>
        <div className={css.header}>
          <div className={css.title}>DELETE DEVICE TYPE</div>
          <div className="pt4 pb4">
            Are you sure you want to delete {deviceType.name}?
          </div>
        </div>
        <LinkReplace to={pathname}>
          <button className={css.btnCancel}>Cancel</button>
        </LinkReplace>
        <LinkReplace to={pathname}>
          <button
            className={`${css.btnDelete} ml2`}
            onClick={() => {

              this.delete(deviceType.id)

            }}
          >
            Delete
          </button>
        </LinkReplace>
      </div>
    )

  }

}

export default DeleteDeviceType
