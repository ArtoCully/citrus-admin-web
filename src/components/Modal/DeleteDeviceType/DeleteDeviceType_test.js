// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import DeleteDeviceType from './DeleteDeviceType'


const mockProps = {
  location: { pathname: '/', state: { deviceType: {} } },
  deviceType: {},
}


it('<DeleteDeviceType> renders the DeleteDeviceType content', () => {

  const wrapper = shallow(
    <DeleteDeviceType
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
