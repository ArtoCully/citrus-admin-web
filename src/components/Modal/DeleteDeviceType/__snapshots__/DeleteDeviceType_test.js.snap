// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteDeviceType> renders the DeleteDeviceType content 1`] = `
<div
  className="deleteDeviceType"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      DELETE DEVICE TYPE
    </div>
    <div
      className="pt4 pb4"
    >
      Are you sure you want to delete 
      ?
    </div>
  </div>
  <LinkReplace
    to="/"
  >
    <button
      className="btnCancel"
    >
      Cancel
    </button>
  </LinkReplace>
  <LinkReplace
    to="/"
  >
    <button
      className="btnDelete ml2"
      onClick={[Function]}
    >
      Delete
    </button>
  </LinkReplace>
</div>
`;
